packages/web-runtime/src/components/ApplicationsMenu.vue

```diff
@@ -21,11 +21,22 @@
     >
       <ul class="uk-grid-small uk-text-center" uk-grid>
         <li v-for="(n, nid) in menuItems" :key="`apps-menu-${nid}`" class="uk-width-1-3">
-          <a v-if="n.url" key="apps-menu-external-link" :target="n.target" :href="n.url">
+          <a
+            v-if="n.url"
+            id="runtime-apps-menu-external-link"
+            key="apps-menu-external-link"
+            :target="n.target"
+            :href="n.url"
+          >
             <oc-icon :name="n.iconMaterial" size="xlarge" />
             <span class="uk-display-block" v-text="$gettext(n.title)" />
           </a>
-          <router-link v-else key="apps-menu-internal-link" :to="n.path">
+          <router-link
```

Comment:
same here with the id

Reply:
@paulcod3 I have used `data-testid`. will that work?

Reply:
@saw-jan nope, same problem. An ID should always unique (hence the name, "unique identifier"). You could add a class or get funky with JS by concatenating the ID in a `"apps-menu-external-link-" + link-id` string but let's think of a different solution here